src/test/java/com/google/step/datamanager/RestaurantManagerDatastoreTest.java

```diff
@@ -0,0 +1,111 @@
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public final class RestaurantManagerDatastoreTest {
+
+  private static final String RESTAURANT_NAME_A = "A";
+  private static final String BLOBKEY_A = "a_blob_key";
+
+  private static final String RESTAURANT_NAME_B = "B";
+  private static final String BLOBKEY_B = "b_blob_key";
+
+  private final LocalServiceTestHelper helper =
+      new LocalServiceTestHelper(
+          new LocalDatastoreServiceTestConfig().setApplyAllHighRepJobPolicy());
+
+  private final RestaurantManagerDatastore restaurantManagerDatastore =
+      new RestaurantManagerDatastore();
+
+  @Before
+  public void setUp() {
+    helper.setUp();
+  }
+
+  @After
+  public void tearDown() {
+    helper.tearDown();
+  }
+
+  @Test
+  public void testCreateRestaurant_success() throws Exception {
+    Restaurant restaurant =
+        restaurantManagerDatastore.createRestaurant(RESTAURANT_NAME_A, BLOBKEY_A);
+    assertEquals(RESTAURANT_NAME_A, restaurant.name);
+    assertEquals(BLOBKEY_A, restaurant.photoBlobkey);
+  }
+
+  @Test
+  public void testReadRestaurant_success() throws Exception {
+    Restaurant restaurant_A =
+        restaurantManagerDatastore.createRestaurant(RESTAURANT_NAME_A, BLOBKEY_A);
+    Restaurant restaurant_A_Test = restaurantManagerDatastore.readRestaurant(restaurant_A.id);
+    assertEquals(RESTAURANT_NAME_A, restaurant_A_Test.name);
+    assertEquals(BLOBKEY_A, restaurant_A_Test.photoBlobkey);
+  }
+
+  @Test
+  public void testReadRestaurant_notExist() throws Exception {
+    Restaurant restaurant = restaurantManagerDatastore.readRestaurant(1000);
+    assertNull(restaurant);
+  }
+
+  @Test
+  public void testUpdateRestaurantName() throws Exception {
+    Restaurant restaurant_A =
+        restaurantManagerDatastore.createRestaurant(RESTAURANT_NAME_A, BLOBKEY_A);
+    Restaurant restaurant_A_New = new Restaurant(restaurant_A.id, RESTAURANT_NAME_B, BLOBKEY_A);
+    Restaurant restaurant_A_Updated = restaurantManagerDatastore.updateRestaurant(restaurant_A_New);
+    assertEquals(RESTAURANT_NAME_B, restaurant_A_Updated.name);
+    assertEquals(BLOBKEY_A, restaurant_A_Updated.photoBlobkey);
+  }
+
+  @Test
+  public void testUpdateRestaurantNameNull() throws Exception {
+    Restaurant restaurant_A =
+        restaurantManagerDatastore.createRestaurant(RESTAURANT_NAME_A, BLOBKEY_A);
+    Restaurant restaurant_A_New = new Restaurant(restaurant_A.id, null, BLOBKEY_A);
+    Restaurant restaurant_A_Updated = restaurantManagerDatastore.updateRestaurant(restaurant_A_New);
+    assertEquals(RESTAURANT_NAME_A, restaurant_A_Updated.name);
+    assertEquals(BLOBKEY_A, restaurant_A_Updated.photoBlobkey);
+  }
+
+  @Test
+  public void testUpdateRestaurant_blobKey() throws Exception {
+    Restaurant restaurant_A =
+        restaurantManagerDatastore.createRestaurant(RESTAURANT_NAME_A, BLOBKEY_A);
+    Restaurant restaurant_A_New = new Restaurant(restaurant_A.id, RESTAURANT_NAME_A, BLOBKEY_B);
```

Comment:
you can set name to be null here too, then you don't need the `testUpdateRestaurantNameNull()` test